pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -756,6 +830,62 @@ func (c *Controller) GetProxyServiceInstances(proxy *model.Proxy) ([]*model.Serv
 	return out, nil
 }
 
+// ForeignServiceInstanceHandler defines the handler for service instances generated by other registries
+func (c *Controller) ForeignServiceInstanceHandler(si *model.ServiceInstance, event model.Event) {
+	// ignore malformed workload entries. And ignore any workload entry that does not have a label
+	// as there is no way for us to select them
+	if si.Service == nil || si.Service.Attributes.Namespace == "" || len(si.Endpoint.Labels) == 0 {
+		return
+	}
+	var redundantEventForInstance bool
+	// this is from a workload entry. Store it in separate map so that
+	// the InstancesByPort can use these as well as the k8s pods.
+	c.Lock()
+	switch event {
+	case model.EventDelete:
+		delete(c.foreignRegistryInstancesByIP, si.Endpoint.Address)
+	default: // add or update
+		if old, exists := c.foreignRegistryInstancesByIP[si.Endpoint.Address]; exists {
+			// ignore spurious updates
+			if model.CompareForeignServiceInstances(old, si) {
```

Comment:
For simplicity, we can always update it here. And leave the WorkloadEntry update check in the real registry(serviceentry).